lib/grp.gi

```diff
@@ -4456,14 +4449,45 @@ end );
 #M  GroupByGenerators( <gens>, <id> )
 ##
 InstallMethod( GroupByGenerators,
-    "delegate to `GroupWithGenerators'",
+    "replace domain by suitable list, and call `GroupWithGenerators'",
     [ IsCollection ],
+    function( gens )
+    if HasGeneratorsOfMagmaWithInverses( gens ) then
+      gens:= GeneratorsOfMagmaWithInverses( gens );
+    elif HasGeneratorsOfMagmaWithOne( gens ) then
+      gens:= GeneratorsOfMagmaWithOne( gens );
+    elif HasGeneratorsOfMagma( gens ) then
+      gens:= GeneratorsOfMagma( gens );
```

Comment:
Hmm. If we do this for GroupByGenerators, shouldn't we also add similar methods to all other `*ByGenerators*` functions? But there are dozens of these. So I am rather inclined to go the other direction, and say we shouldn't do this at all...

I found these *ByGenerator operations:
```
AdditiveGroupByGenerators
AdditiveMagmaByGenerators
AdditiveMagmaWithInversesByGenerators
AdditiveMagmaWithZeroByGenerators
AlgebraByGenerators
AlgebraWithOneByGenerators
BiAlgebraModuleByGenerators
DefaultFieldByGenerators
DefaultRingByGenerators
DivisionRingByGenerators
DomainByGenerators
FLMLORByGenerators
FLMLORWithOneByGenerators
FieldByGenerators
FieldOverItselfByGenerators
GroupByGenerators
IdealByGenerators
InducedPcgsByGenerators
InducedPcgsByGeneratorsNC
InverseMonoidByGenerators
InverseSemigroupByGenerators
IsomorphismFpGroupByGenerators
IsomorphismFpGroupByGeneratorsNC
LeftAlgebraModuleByGenerators
LeftIdealByGenerators
LeftMagmaIdealByGenerators
LeftModuleByGenerators
MagmaByGenerators
MagmaIdealByGenerators
MagmaWithInversesByGenerators
MagmaWithOneByGenerators
MonoidByGenerators
NaturalHomomorphismByGenerators
NearAdditiveGroupByGenerators
NearAdditiveMagmaByGenerators
NearAdditiveMagmaWithInversesByGenerators
NearAdditiveMagmaWithZeroByGenerators
OrbitStabilizerListByGenerators
RepresentativeTomByGenerators
RepresentativeTomByGeneratorsNC
RightAlgebraModuleByGenerators
RightIdealByGenerators
RightMagmaIdealByGenerators
RingByGenerators
RingWithOneByGenerators
SemigroupByGenerators
SemigroupIdealByGenerators
SemiringByGenerators
SemiringWithOneAndZeroByGenerators
SemiringWithOneByGenerators
SemiringWithZeroByGenerators
TwoSidedIdealByGenerators
```

Reply:
Perhaps the discussion of issue #2703 was not explicit enough:
The point is just to fix a side-effect (which I would call a bug) of pull request #2522,
which breaks functionality in GAP 4.10 that was available in earlier versions.

Reply:
But the old code never did something like this. It just took the set of elements of the given domain / collection as generators.